### PR TITLE
Bug 1450004 - Migrate Firefox to swift 4 

### PR DESCRIPTION
**Pull request by farhanpatel**
This is a work in progress. I've split this up into logical commits. Trying to make this as easy to follow as possible.

### CODE REVIEW

File: ClientTests/TestHistory.swift [hunk 1]
```diff
@@ -199,7 +199,7 @@ class TestHistory: ProfileTest {
             cb()
             return
         } else {
-            runRandom(&history, cmdIn: -1) { [history]_ in
```
Comment (justindarc) [inline ClientTests/TestHistory.swift]: how was that working?

File: Client/Frontend/Browser/FindInPageBar.swift [hunk 2]
```diff
@@ -112,15 +112,15 @@ class FindInPageBar: UIView {
         searchText.snp.makeConstraints { make in
             make.leading.top.bottom.equalTo(self).inset(UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 0))
         }
-        searchText.setContentHuggingPriority(UILayoutPriorityDefaultLow, for: .horizontal)
-        searchText.setContentCompressionResistancePriority(UILayoutPriorityDefaultLow, for: .horizontal)
+        searchText.setContentHuggingPriority(UILayoutPriority.defaultLow, for: .horizontal)
+        searchText.setContentCompressionResistancePriority(UILayoutPriority.defaultLow, for: .horizontal)
```
[the remark this answers is not in the document]
Comment (farhanpatel) [reply Client/Frontend/Browser/FindInPageBar.swift]: Yup. Will do

File: ClientTests/TestHistory.swift (hunk 1 above)
Comment (justindarc) [reply ClientTests/TestHistory.swift]: what is `[history]_` going to do?

File: Client/Frontend/AuthenticationManager/PasscodeViews.swift [hunk 3]
```diff
@@ -31,11 +31,11 @@ class PasscodeInputView: UIView, UIKeyInput {
     fileprivate var inputtedCode: String = ""
 
     fileprivate var blankDigitString: NSAttributedString {
-        return NSAttributedString(string: "\(blankCharacter)", attributes: [NSFontAttributeName: digitFont])
+        return NSAttributedString(string: "\(blankCharacter)", attributes: [NSAttributedStringKey.font: digitFont])
```
[the remark this answers is not in the document]
Comment (justindarc) [reply Client/Frontend/AuthenticationManager/PasscodeViews.swift]: Oh ok. I wasn't sure either by just looking at it if the compiler could infer that based on the type definition of the `attributes` argument.

File: ClientTests/TestHistory.swift (hunk 1 above)
Comment (justindarc) [reply ClientTests/TestHistory.swift]: I've seen `_` as an empty argument for a closure, but shouldn't there have been a comma? i.e. `[history], _`?

Comment (farhanpatel) [reply ClientTests/TestHistory.swift]: there usually isnt a comma

**Pull request merged**